### PR TITLE
Add plugin to export images

### PR DESCRIPTION
**Pull request by yiqunw700**
Images are downloaded in png format for png and jpg. Gif images will also be downloaded, in the gif format. But there is only one image hash returned by the figma plugin api which is for the gif itself. The REST API however returns an `imageRef` which might be the image hash of the first frame and a `gifRef` which is the image hash returned by the plugin api.

The plugin data is in {image_hash: res_name,...} format.

Right now all images are exported. Will allow user to pick which images to export in a following pr.

Towards: #1556

### CODE REVIEW

File: support-figma/extended-layout-plugin/src/ui.html [hunk 1]
```diff
@@ -385,6 +386,18 @@
 </div>
 <!-- END OF LOCALIZATION PLUGIN UI -->
 
+<!-- BEGIN OF IMAGE RES PLUGIN UI -->
+<div id="imageExport" class="page page-padding-large">
+  <div style="border-bottom: solid #000000; font-weight: bold;">Images:</div>
+  <div style="font-size: 14; margin-top: 8px;">* Jpg or png images are downloaded in png format, gif images are downloaded in gif format but DesignCompose doesn't support gif.</div>
```
Comment (rylin8) [inline support-figma/extended-layout-plugin/src/ui.html:392]: Not sure what the '*' is for at the beginning, perhaps remove it?

File: support-figma/extended-layout-plugin/src/image-res-module.ts [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    imageNodesArray: Array.from(imageHashToNodesMap),
+    imageResNameArray: Array.from(imageHashToResMap),
+  });
+}
+
+async function exportNodeImagesAsync(
+  node: SceneNode,
+  cachedHashToResMap: Map<string, string>
+) {
+  let nodeWithFills: GeometryMixin = node as GeometryMixin;
```
Comment (rylin8) [inline support-figma/extended-layout-plugin/src/image-res-module.ts:82]: What is a GeometryMixin?

File: support-figma/extended-layout-plugin/src/image-res-module.ts [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { json } from "stream/consumers";
+import * as Utils from "./utils";
+
+const IMAGE_HASH_TO_RES_KEY = "image_hash_to_res";
+
+// Hash to image bytes map
+const imageBytesMap = new Map<string, Uint8Array>();
```
Comment (rylin8) [inline support-figma/extended-layout-plugin/src/image-res-module.ts:22]: Can you rename this to `imageHashToBytesMap` to be consistent?

File: support-figma/extended-layout-plugin/manifest.json [hunk 4]
```diff
@@ -16,14 +16,17 @@
...
+        { "name": "Export Images", "command": "export-images" },
+        { "name": "Clear Image Resource", "command": "clear-image-res" }
       ]
     }
   ],
   "networkAccess": {
     "allowedDomains": [
       "https://cdn.jsdelivr.net/gh/thomas-lowry/figma-plugin-ds/dist/figma-plugin-ds.css",
-      "https://rsms.me"
+      "https://rsms.me",
```
[the remark this answers is not in the document]
Comment (yiqunw700) [reply support-figma/extended-layout-plugin/manifest.json:28]: The https://cdn.jsdelivr.net/gh/thomas-lowry/figma-plugin-ds/dist/figma-plugin-ds.css uses Inter fonts. "https://rsms.me" is for the Inter fonts.

File: support-figma/extended-layout-plugin/src/ui.html (hunk 1 above)
Comment (yiqunw700) [reply support-figma/extended-layout-plugin/src/ui.html:392]: Those are instructions or explanations of the plugin. I changed it to <li></li>.

File: support-figma/extended-layout-plugin/src/image-res-module.ts (hunk 3 above)
Comment (yiqunw700) [reply support-figma/extended-layout-plugin/src/image-res-module.ts]: Done.

File: support-figma/extended-layout-plugin/src/image-res-module.ts (hunk 2 above)
Comment (yiqunw700) [reply support-figma/extended-layout-plugin/src/image-res-module.ts:82]: `interface GeometryMixin extends MinimalStrokesMixin, MinimalFillsMixin`. It is like the base interface for nodes both have strokes and fills.
I broke it into MinimalStrokesMixin and MinimalFillsMixin.

**Review by rylin8 — APPROVED** [review]

**Pull request merged**